launcher/java/JavaUtils.cpp

```diff
@@ -441,13 +441,13 @@ QString JavaUtils::getJavaCheckPath()
 
 QStringList getMinecraftJavaBundle()
 {
-    QString partialPath;
     QString executable = "java";
     QStringList processpaths;
 #if defined(Q_OS_OSX)
-    partialPath = FS::PathCombine(QDir::homePath(), "Library/Application Support");
+    processpaths << FS::PathCombine(QDir::homePath(), FS::PathCombine("Library", "Application Support", "minecraft", "runtime"));
 #elif defined(Q_OS_WIN32)
-    partialPath = QProcessEnvironment::systemEnvironment().value("LOCALAPPDATA", "");
+    QString partialPath = QProcessEnvironment::systemEnvironment().value("LOCALAPPDATA", "");
```

Comment:
Why are we using LOCALAPPDATA
That would make it AppData\Local\.minecraft when it is located in AppData\Roaming|.minecraft

Reply:
huh? that does exist for the non ms store launcher
it's just it was detecting it from AppData\Local which never exists